QuestionAnswering/QA_ICL/backend/fastapi_app/README.md

```diff
@@ -54,40 +54,44 @@ ontomops_fileserver:
 
 Precedence: `app.local.yaml` > `app.{APP_ENV}.yaml` > `app.yaml`.
 
+
 ## Native installation
 
 ### Prerequisites
 
-- (recommended) Linux enviroment, including WSL
+- (recommended) Linux enviroment
 - python>=3.8
 - [conda](https://conda.io/projects/conda/en/latest/user-guide/install/index.html)
-
   
 ### Steps
 
+1. Follow the points in the section [Initial setup](initial-setup) to ensure that the required services and resources are properly configured.
+
 1. Create conda environment and activate it.
    ```{bash}
    conda create --name qa_backend python==3.10
    conda activate qa_backend
    ```
 
-2. Install dependencies.
+1. Install dependencies.
    ```{bash}
    pip install -r requirements.txt
    ```
 
-3. Ingest required resources into Redis server, as per the section ['Required resources'](#required-resources).
+1. Ingest required resources into Redis server, as per the section ['Required resources'](#required-resources).
 
-4. Start the server:
+1. Start the server:
    - In debug mode (app is automatically reloaded upon code changes), `uvicorn main:app --reload --log-config=log_conf.yaml`.
    - In production mode, `uvicorn main:app --host=0.0.0.0 --log-config log_conf.yaml --workers 4`.
-   - The app will be available at `localhost:8000`. 
+   - The app will be available at `localhost:8000`.  
+   <!-- TODO: add a health check endpoint -->
    - To expose the app at a different port, use the command line argument `--port {port}` e.g. `uvicorn main:app --reload --log-config=log_conf.yaml --port 5000`.
 
-5. Whenever any resource needs to be updated, re-run the ingestion script for that specific resource with the argument `--drop_index --invalidate_cache` to (1) trigger Redis to flush the old index and create a new one, and (2) re-create on-disk cache for the processed resource. KIV: allow user to add new resource entries without processing everything again and recreatng the index.
+1. Whenever any dataset needs to be updated, re-run the ingestion script for that specific dataset with the argument `--drop_index --invalidate_cache` to (1) trigger Redis to flush the old index and create a new one, and (2) re-create on-disc cache for the processed datasets. KIV: allow user to add new resource entries without processing everything again and recreatng the index.
 
-## Dockerised installation
+## Docker installation
 
+1. Follow the points in the section [Initial setup](initial-setup) to ensure that the required services and resources are properly configured.
```

Comment:
> Incorrect link

I have updated the link.

> Maybe just add the steps here, because we should document the core component and how to deploy it

Do you mean to move the content of the section "Initial setup" into the "Steps" section? If so, it will be harder to reference the instructions in "Initial setup" in the "Docker installation" instructions. Or do you mean to briefly repeat what is said in "Initial setup" in "Steps"?

Reply:
Hmm I see the issue. I suggest to edit the step 1 sentence to Deploy the required services, add the resources and configure the parameters as defined in [this section](#initial-setup)

On this note, in the required services section, split them up to what should be deployed and what is just an API call to a publicly available web services. The former  are things we have to set up ourselves, and the latter are things we can call without requiring set up

Reply:
I have made changes, please review the latest fastapi_readme
- Restructure Required services section into Services to Deploy and External API Services
- Clarify deployment instructions for core components